src/command.rs

```diff
@@ -0,0 +1,306 @@
+// (c) Copyright 2023 Helsing GmbH. All rights reserved.
+
+use async_recursion::async_recursion;
+use eyre::ContextCompat;
+
+use crate::{
+    credentials::Credentials,
+    generator,
+    lock::{LockedPackage, Lockfile},
+    manifest::{Dependency, Manifest, PackageManifest},
+    package::{DependencyGraph, PackageName, PackageStore, PackageType},
+    registry::{Artifactory, ArtifactoryConfig, Registry},
+    Language,
+};
+
+use std::{env, path::Path};
+
+use eyre::{ensure, Context};
+use semver::{Version, VersionReq};
+
+/// Initializes the project
+pub async fn init(r#type: PackageType, name: Option<PackageName>) -> eyre::Result<()> {
+    ensure!(
+        !Manifest::exists().await?,
+        "Cant initialize existing project"
+    );
+
+    const DIR_ERR: &str = "Failed to read current directory name";
+
+    let name = match name {
+        Some(name) => name,
+        None => std::env::current_dir()?
+            .file_name()
+            .wrap_err(DIR_ERR)?
+            .to_str()
+            .wrap_err(DIR_ERR)?
+            .parse()?,
+    };
+
+    const VERSION: Version = Version::new(0, 1, 0);
+
+    let manifest = Manifest {
+        package: PackageManifest {
+            r#type,
+            name,
+            version: VERSION,
+            description: None,
+        },
+        dependencies: vec![],
+    };
+
+    manifest.write().await?;
+
+    PackageStore::create().await
+}
+
+/// Adds a dependency to this project
+pub async fn add(dependency: String) -> eyre::Result<()> {
+    let lower_kebab = |c: char| (c.is_lowercase() && c.is_ascii_alphabetic()) || c == '-';
+
+    let (repository, dependency) = dependency
+        .trim()
+        .split_once('/')
+        .wrap_err("Invalid dependency specification")?;
+
+    ensure!(
+        repository.chars().all(lower_kebab),
+        "Repositories must be in lower kebab case"
+    );
+
+    let (package, version) = dependency
+        .split_once('@')
+        .wrap_err_with(|| format!("Invalid dependency specification: {dependency}"))?;
+
+    let package = package
+        .parse::<PackageName>()
+        .wrap_err_with(|| format!("Invalid package name supplied: {package}"))?;
+
+    let version = version
+        .parse::<VersionReq>()
+        .wrap_err_with(|| format!("Invalid version requirement supplied: {package}"))?;
+
+    let mut manifest = Manifest::read().await?;
+
+    manifest
+        .dependencies
+        .push(Dependency::new(repository.to_owned(), package, version));
+
+    manifest.write().await
```

Comment:
should add and remove also be updating the lockfile?

Reply:
I thought about that, and I think yes, ideally. But since we're already doing substantive changes here I'm in favour of following up on that in a later PR.
@mara-schulke FYSA

Reply:
Created #60 to track. Resolving for now (but happy to unresolve if Mara thinks this is worth scoping now).